src/main/scala/com/databricks/spark/xml/DefaultSource.scala

```diff
@@ -90,9 +89,7 @@ class DefaultSource
     }
     if (doSave) {
       // Only save data when the save mode is not ignore.
-      val codecClass =
-        CompressionCodecs.getCodecClass(XmlOptions(parameters).codec)
-      data.saveAsXmlFile(filesystemPath.toString, parameters, codecClass)
+      XmlFile.saveAsXmlFile(data, filesystemPath.toString, parameters)
```

Comment:
Here, I moved the codec handling inside to `saveAsXmlFile` so that `codec` option in `parameters` can be also concerned.